woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -705,58 +705,142 @@ def test_make_index(sample_df):
 
-    sample_df = sample_df.sort_values('full_name')
-    assert type(sample_df.index) == pd.Int64Index
-    sample_df.ww.init()
-    assert type(sample_df.index) == pd.RangeIndex
+    sorted_df = schema_df.ww.sort_values('full_name')
+    assert sorted_df.ww.index is None
+    pd.testing.assert_index_equal(pd.Int64Index([88, 77, 99, 66]), sorted_df.index)
 
 
-def test_underlying_index(sample_df):
+def test_underlying_index_set(sample_df):
     if dd and isinstance(sample_df, dd.DataFrame):
         pytest.xfail('Setting underlying index is not supported with Dask input')
     if ks and isinstance(sample_df, ks.DataFrame):
         pytest.xfail('Setting underlying index is not supported with Koalas input')
 
-    specified_index = pd.Index
-    unspecified_index = pd.RangeIndex
-
+    # Sets underlying index at init
     schema_df = sample_df.copy()
     schema_df.ww.init(index='full_name')
     assert 'full_name' in schema_df.columns
     assert schema_df.index.name is None
     assert (schema_df.index == schema_df['full_name']).all()
-    assert type(schema_df.index) == specified_index
 
+    # Sets underlying index on update
     schema_df = sample_df.copy()
     schema_df.ww.init(index='id')
     schema_df.ww.set_index('full_name')
+    assert schema_df.ww.index == 'full_name'
     assert 'full_name' in schema_df.columns
     assert (schema_df.index == schema_df['full_name']).all()
     assert schema_df.index.name is None
-    assert type(schema_df.index) == specified_index
 
-    # test removing index removes the dataframe's index
+    # confirm removing Woodwork index doesn't change underlying index
     schema_df.ww.set_index(None)
-    assert type(schema_df.index) == unspecified_index
+    assert schema_df.ww.index is None
+    assert (schema_df.index == schema_df['full_name']).all()
 
+    # Sets underlying index for made index
     schema_df = sample_df.copy()
     schema_df.ww.init(index='made_index', make_index=True)
     assert 'made_index' in schema_df
     assert (schema_df.index == [0, 1, 2, 3]).all()
     assert schema_df.index.name is None
-    assert type(schema_df.index) == specified_index
+
+
+def test_underlying_index_reset(sample_df):
+    if dd and isinstance(sample_df, dd.DataFrame):
+        pytest.xfail('Setting underlying index is not supported with Dask input')
+    if ks and isinstance(sample_df, ks.DataFrame):
+        pytest.xfail('Setting underlying index is not supported with Koalas input')
+
+    specified_index = pd.Index
+    unspecified_index = pd.RangeIndex
+
+    sample_df.ww.init()
+    assert type(sample_df.index) == unspecified_index
+
+    sample_df.ww.set_index('full_name')
+    assert type(sample_df.index) == specified_index
+
+    sample_df.ww.set_index(None)
+    assert type(sample_df.index) == specified_index
+
+    # Use pandas operation to reset index
+    sample_df.ww.reset_index(drop=True, inplace=True)
+    assert type(sample_df.index) == unspecified_index
```

Comment:
Do you think it would be worth confirming the behavior of `df.ww.reset_index` when we have an index set, or when we use `inplace=False`? Not sure it is necessary though...

Reply:
yeah, happy to add both of those checks